src/containers/datasets/contextual-layers/basemaps/index.ts

```diff
@@ -1,16 +1,28 @@
-const BASEMAPS = [
+export type BasemapId = 'light' | 'dark' | 'satellite';
+
+type BasemapTypes = {
+  id: BasemapId;
+  type: 'basemap';
+  name: string;
+  url: string;
+};
+
+const BASEMAPS: BasemapTypes[] = [
```

Comment:
Add `satisfies BasemapTypes[]` at the end of the array. Otherwise, the array is not actually typed (just inferring the type) and you can fill it with wrong properties/values.